backend/src/main/scala/bloop/logging/BloopLogger.scala

```diff
@@ -56,13 +57,13 @@ final class BloopLogger(
   }
 
   private def print(msg: String, fn: String => Unit): Unit = {
-    val lines = msg.split("\\r?\\n", -1)
+    val lines = msg.splitLines // reliable line splitter
     val printFn = if (colorOutput) fn else (str: String) => fn(BloopLogger.stripColors(str))
     lines.foreach(printFn)
   }
 
   private def printInfo(line: String): Unit = {
-    out.println(line)
+    out.print(line + EOL) // end-of-line matches shell environments.
```

Comment:
I just pushed changes incorporating all your suggestions.  
I simplified `io.Environment` by removing `EOL`, the tests for `OSTYPE`, and some other simplifications, reducing code by about 50 lines.
I also found some additional references to `System.lineSeparator`, including two places that don't link with `io.Environment`, so I had to add local definitions for `lineSeparator` and the `splitLines` regex.

```diff
diff --git a/frontend/src/test/scala/bloop/bsp/BspClientTest.scala b/frontend/src/test/scala/bloop/bsp/BspClientTest.scala
-                    .replace(System.lineSeparator, " ")
+                    .replace("\n|\r\n|\r", " ")

diff --git a/launcher/src/main/scala/bloop/launcher/package.scala b/launcher/src/main/scala/bloop/launcher/package.scala
-        .split(System.lineSeparator())
+        .split("\n|\r\n|\r", -1)
-        .mkString(System.lineSeparator()),
+        .mkString(lineSeparator),
+
+  /**
+   * SHELL path implies preference for '\n' instead of Windows default.
+   * @return '\n' if SHELL path recognized, system line.separator otherwise.
+   */
+  lazy val lineSeparator = Option(System.getenv("SHELL")) match {
+    case Some(sh) if sh.toLowerCase.matches("/.*/bin/[a-z]*sh(.exe)") => "\n"
+    case _ => System.getProperty("line.separator", "\n")
+  }
```